### PR TITLE
[DPE-5531] Charmed Spark Hardening guide

### PR DESCRIPTION
**Pull request by deusebio**
Hardening user-guide, to complement the security documentation of SSDLC. The documention here focuses more on the guidance to achieve a secure deployment, by providing a number of links to how-to guides (internal and external documentation)

@welpaolo @theoctober19th

### CODE REVIEW

File: releases/3.4/yaml/docs/explanation/e-hardening.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+Charmed Spark K8s operators, including Spark History server, Kyuubi, and Integration Hub, install a pinned revision of the 
+Charmed Spark images outlined in the previous table in order to provide reproducible and secure environments. 
+New versions of Charmed Spark K8s operators may therefore be released to provide patching of vulnerabilities (CVEs). 
+It is important to refresh the charm regularly to make sure the workload is as secure as possible. 
+For more information on how to refresh the charm, see the [how-to upgrade](TODO) guide.
+
+### Encryption
+
+We recommend to deploy Charmed Spark K8s with encryption enabled for the communication between components whenever available and supported.
```
Comment (izmalk) [inline releases/3.4/yaml/docs/explanation/e-hardening.md:101]: Looks like something is missing here. Maybe a verb after components?

File: releases/3.4/yaml/docs/explanation/e-hardening.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+In the following, we provide further information on how to encrypt the various data flow between the different components of the solution:
+
+* Client <> Kubernetes API connections
+* Object Storage connections
```
Comment (izmalk) [inline releases/3.4/yaml/docs/explanation/e-hardening.md:105]:
```suggestion
* Object storage connections
```

File: releases/3.4/yaml/docs/explanation/e-hardening.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+#### Kyuubi JDBC endpoint
+
+Authentication can be enabled for Kyuubi via its integration with PostgreSQL charm on the `auth-db` interface. Currently only one admin user is enabled, whose credentials can be retrieved 
+using the [`get-jdbc-endpoint` action](https://charmhub.io/kyuubi-k8s/actions#get-jdbc-endpoint).
+
+### Monitoring and Auditing
+
+Charmed Spark provides native integration with the [Canonical Observability Stack (COS)](https://charmhub.io/topics/canonical-observability-stack).
+To reduce the blast radius of infrastructure disruptions, the general recommendation is to deploy COS and the observed application into 
+separate environments, isolated one another. Refer to the [COS production deployments best practices](https://charmhub.io/topics/canonical-observability-stack/reference/best-practices)
```
Comment (izmalk) [inline releases/3.4/yaml/docs/explanation/e-hardening.md:189]:
```suggestion
separate environments, isolated from one another. Refer to the [COS production deployments best practices](https://charmhub.io/topics/canonical-observability-stack/reference/best-practices) page
```

File: releases/3.4/yaml/docs/explanation/e-hardening.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+for more information. 
+
+Refer to How-To user guide for more information on [how to integrate and customize](/t/charmed-spark-k8s-documentation-enable-monitoring/13063)
```
Comment (izmalk) [inline releases/3.4/yaml/docs/explanation/e-hardening.md:192]:
```suggestion
For more information on how to enable and customise monitoring with COS, see the [Guide](/t/charmed-spark-k8s-documentation-enable-monitoring/13063).
```

File: releases/3.4/yaml/docs/explanation/e-hardening.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+### Monitoring and Auditing
+
+Charmed Spark provides native integration with the [Canonical Observability Stack (COS)](https://charmhub.io/topics/canonical-observability-stack).
+To reduce the blast radius of infrastructure disruptions, the general recommendation is to deploy COS and the observed application into 
+separate environments, isolated one another. Refer to the [COS production deployments best practices](https://charmhub.io/topics/canonical-observability-stack/reference/best-practices)
+for more information. 
+
+Refer to How-To user guide for more information on [how to integrate and customize](/t/charmed-spark-k8s-documentation-enable-monitoring/13063) 
+the Charmed Spark deployment with COS.
```
Comment (izmalk) [inline releases/3.4/yaml/docs/explanation/e-hardening.md:193]:
```suggestion
```

File: releases/3.4/yaml/docs/explanation/e-hardening.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+To reduce the blast radius of infrastructure disruptions, the general recommendation is to deploy COS and the observed application into 
+separate environments, isolated one another. Refer to the [COS production deployments best practices](https://charmhub.io/topics/canonical-observability-stack/reference/best-practices)
+for more information. 
+
+Refer to How-To user guide for more information on [how to integrate and customize](/t/charmed-spark-k8s-documentation-enable-monitoring/13063) 
+the Charmed Spark deployment with COS. 
+
+## Additional Resources
+
+For further information and details on the security and cryptographic specifications used by Charmed Spark, please refer to the [Security Explanation page](TODO).
```
Comment (izmalk) [inline releases/3.4/yaml/docs/explanation/e-hardening.md:197]: Add the valid link here

File: releases/3.4/yaml/docs/explanation/e-hardening.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+
+Charmed Spark provides native integration with the [Canonical Observability Stack (COS)](https://charmhub.io/topics/canonical-observability-stack).
+To reduce the blast radius of infrastructure disruptions, the general recommendation is to deploy COS and the observed application into 
+separate environments, isolated one another. Refer to the [COS production deployments best practices](https://charmhub.io/topics/canonical-observability-stack/reference/best-practices)
+for more information. 
+
+Refer to How-To user guide for more information on [how to integrate and customize](/t/charmed-spark-k8s-documentation-enable-monitoring/13063) 
+the Charmed Spark deployment with COS. 
+
+## Additional Resources
```
Comment (izmalk) [inline releases/3.4/yaml/docs/explanation/e-hardening.md:195]:
```suggestion
## Additional resources
```

File: releases/3.4/yaml/docs/explanation/e-hardening.md (hunk 6 above)
Comment (deusebio) [reply releases/3.4/yaml/docs/explanation/e-hardening.md]: Oh, yes. When I initially wrote this, the security docs was not out yet. Updated! Nice catch!

**Review by izmalk — APPROVED** [review]

**Pull request closed without merging**